src/core/Form/SearchInput/SearchInput.tsx

```diff
@@ -1,47 +1,265 @@
+  /** Input name */
+  name?: string;
+  /** Set components width to 100% */
+  fullWidth?: boolean;
+  /** Callback for input text change */
+  onChange?: (value: StateValue) => void;
+  /** Callback for onBlur event */
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+  /** Callback for search button click */
+  onSearch?: (value: StateValue) => void;
+}
 
 const baseClassName = 'fi-search-input';
-const iconBaseClassName = `${baseClassName}_icon`;
+const searchInputClassNames = {
+  fullWidth: `${baseClassName}--full-width`,
+  error: `${baseClassName}--error`,
+  notEmpty: `${baseClassName}--not-empty`,
+  inputElement: `${baseClassName}_input`,
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  functionalityContainer: `${baseClassName}_functionality-container`,
+  button: `${baseClassName}_button`,
+  searchButton: `${baseClassName}_button-search`,
+  searchIcon: `${baseClassName}_button-search-icon`,
+  clearButton: `${baseClassName}_button-clear`,
+  clearIcon: `${baseClassName}_button-clear-icon`,
+};
 
-export interface SearchInputProps extends Omit<TextInputProps, 'status'> {
-  status?: 'default' | 'error';
+interface SearchInputState {
+  value: string | number | string[] | undefined;
+}
+
+class BaseSearchInput extends Component<SearchInputProps> {
+  state: SearchInputState = {
+    value: this.props.value || this.props.defaultValue || '',
+  };
+
+  private inputRef = createRef<HTMLInputElement>();
+
+  private id: string;
+
+  private statusTextId: string;
+
+  static getDerivedStateFromProps(
+    nextProps: SearchInputProps,
+    prevState: SearchInputState,
+  ) {
+    const { value } = nextProps;
+    if ('value' in nextProps && value !== prevState.value) {
+      return { value };
+    }
+    return null;
+  }
+
+  constructor(props: SearchInputProps) {
+    super(props);
+    this.id = `${idGenerator(props.id)}`;
+    this.statusTextId = `${this.id}-statusText`;
+  }
+
+  render() {
+    const {
+      value,
+      defaultValue,
+      className,
+      labelText,
+      labelMode,
+      clearButtonLabel,
+      searchButtonLabel,
+      searchButtonProps,
+      inputContainerProps,
+      onChange: propOnChange,
+      onSearch: propOnSearch,
+      children,
+      status,
+      statusText,
+      visualPlaceholder,
+      id,
+      fullWidth,
+      'aria-describedby': ariaDescribedBy,
+      ...passProps
+    } = this.props;
+
+    const conditionalSetState = (newValue: StateValue) => {
+      if (!('value' in this.props)) {
+        this.setState({ value: newValue });
+      }
+      if (propOnChange) {
+        propOnChange(newValue);
+      }
+    };
+
+    const onSearch = () => {
+      if (!!propOnSearch) {
+        propOnSearch(this.state.value);
+      }
+    };
+
+    const onClear = () => {
+      conditionalSetState('');
+      setTimeout(() => {
+        if (this.inputRef.current) {
+          this.inputRef.current.focus();
+        }
+      }, 100);
```

Comment:
Is the chosen `100ms` some standard or why that amount is chosen? 
Just thought that are there some cases that it should be higher to work correctly 🤔

Reply:
It's just a guess. So small that it's barely noticeable, but leaves enough time for re-rendering the component after state change (clear) before moving focus. This way screen readers read the new value rather than what was in the input before clearing.

There would probably be a better way for doing this, but that involves managed state transitions and probably some boolean logic to move focus to input with useLayoutEffect if a clear flag has been set => This kind of state transitioning would require thorough planning and a uniform way for handling for the whole library and is not worth it for this case in my opinion. If something fails, the focus is either not moved, or the old value could be read - so nothing should break in any case.

Reply:
We also discussed this case separately and we did not see reasons to make changes to the current implementation for now.